File: executable-operator/executable-sparksql/src/main/java/fdu/daslab/executable/sparksql/utils/SparkInitUtil.scala
package fdu.daslab.executable.sparksql.utils

import org.apache.spark.SparkContext
import org.apache.spark.sql.SparkSession

/**
 * 初始化spark需要的一些方法
 *
 * @author 刘丰艺
 * @since 2020/10/8 5:30 PM
 * @version 1.0
 */
object SparkInitUtil {

  private val sparkContext = SparkSession.builder()
    .master("local").appName("SparkSQLStage").getOrCreate()
    .sparkContext

  private val sparkSession = SparkSession.builder()
    .master("local").appName("SparkSQLStage").getOrCreate()

  def getDefaultSparkContext(): SparkContext = {
    sparkContext
  }

  def getDefaultSparkSession(): SparkSession = {
    sparkSession
  }
}

File: executable-operator/executable-sparksql/src/main/java/fdu/daslab/executable/sparksql/utils/SparkInitUtil.java
package fdu.daslab.executable.sparksql.utils;

import org.apache.spark.sql.SparkSession;

/**
 * 初始化spark
 *
 * @author 刘丰艺
 * @since 2020/10/27 5:30 PM
 * @version 1.0
 */
public class SparkInitUtil {
    private static SparkSession sparkSession;

    public static SparkSession getDefaultSparkSession() {
        if (sparkSession == null) {
            sparkSession = SparkSession.builder().master("local").appName("SparkSQLStage").getOrCreate();
        }
        return sparkSession;
    }

    public static SparkSession setSparkSession(String master, String appName) {
        sparkSession = SparkSession.builder().master(master).appName(appName).getOrCreate();
        return sparkSession;
    }
}
